api/java/org/openyolo/api/RetrieveRequest.java

```diff
@@ -47,59 +44,53 @@
 import org.openyolo.proto.CredentialRetrieveRequest;
 
 /**
- * A request for credentials, to be sent to credential providers on the device. A request specifies
- * at least one {@link AuthenticationDomain authentication domain} for which a credential is
- * required.
+ * A request for credentials, to be sent to credential providers on the device.
  */
 public class RetrieveRequest implements Parcelable {
 
     /**
      * Parcelable reader for {@link RetrieveRequest} instances.
+     *
      * @see android.os.Parcelable
      */
     public static final Creator<RetrieveRequest> CREATOR = new RetrieveRequestCreator();
 
-    /**
-     * Creates a basic retrieve request for the current application.
-     */
-    public static RetrieveRequest forSelf(Context context) {
-        return new RetrieveRequest.Builder(AuthenticationDomain.getSelfAuthDomain(context))
-                .build();
-    }
-
-    @NonNull
-    private final Set<AuthenticationDomain> mAuthDomains;
-
     @NonNull
     private final Set<Uri> mAuthMethods;
 
     @NonNull
     private final Map<String, byte[]> mAdditionalParams;
 
     private RetrieveRequest(
-            @NonNull Set<AuthenticationDomain> authDomains,
             @NonNull Set<Uri> authMethods,
             @NonNull Map<String, byte[]> additionalParams) {
-        mAuthDomains = authDomains;
         mAuthMethods = authMethods;
         mAdditionalParams = additionalParams;
     }
 
     /**
-     * The set of authentication domains for which a credential is desired. It is the
-     * responsibility of the credential provider to validate that the requester is permitted to
-     * retrieve credentials for this set of domains, and how to interpret requests for domains
-     * which are beyond the provably associated scope of the requester.
+     * Creates a {@link RetrieveRequest} from the given authentication methods.
      */
-    @NonNull
-    public Set<AuthenticationDomain> getAuthenticationDomains() {
-        return mAuthDomains;
+    public static RetrieveRequest forAuthenticationMethods(
+            @NonNull Set<Uri> authenticationMethods) {
+        return new RetrieveRequest.Builder()
+                .setAuthenticationMethods(authenticationMethods)
+                .build();
     }
 
     /**
-     * The set of authentication methods that the requestor supports. This is used to filter the
-     * set of credentials saved by the provider. If no authentication methods are specified, then
-     * no filtering of saved credentials will occur.
+     * Creates a {@link RetrieveRequest} from the given authentication methods.
+     */
+    public static RetrieveRequest forAuthenticationMethods(@NonNull Uri... authenticationMethods) {
```

Comment:
Ack. The codebase currently uses this style.  See line 181: setAuthenticationMethods(Uri... authMethods).